### PR TITLE
224/search bar

### LINKED ISSUE #224: Add a search bar for the spaces grid

### PR DESCRIPTION
**Pull request by pyphilia**
closes #224

### CODE REVIEW

File: src/Home.js [hunk 1]
```diff
@@ -39,15 +41,35 @@ class Home extends Component {
     activity: PropTypes.bool.isRequired,
     history: PropTypes.shape({ length: PropTypes.number.isRequired })
       .isRequired,
+    searchQuery: PropTypes.string.isRequired,
   };
 
-  componentDidMount() {
+  state = {
+    // eslint-disable-next-line react/destructuring-assignment
```
Comment (juancarlosfarah) [inline src/Home.js:48]: There's a way to do this that I think is quite cool and doesn't require disabling this check. It is also more flexible.

https://github.com/graasp/graasp-app-timer/blob/master/src/components/modes/teacher/Settings.js#L52-L57

Comment (juancarlosfarah) [reply src/Home.js:48]: Credit to @solaolateju for the discovery.

File: src/Home.js [hunk 2]
```diff
@@ -39,15 +41,35 @@ class Home extends Component {
...
+    filteredSpaces: this.props.spaces,
+  };
+
+  async componentDidMount() {
     const { dispatchGetSpaces } = this.props;
-    dispatchGetSpaces();
+    await dispatchGetSpaces();
```
Comment (juancarlosfarah) [inline src/Home.js:54]: Do we need to add the await here? I don't think it has any effect. In fact, we can probably remove the `async` from the function. Let me know if that is not the case.

File: src/components/common/Header.js [hunk 3]
```diff
@@ -32,6 +43,9 @@ const Header = ({ classes, handleDrawerOpen, isSidebarOpen }) => {
         >
           <MenuIcon />
         </IconButton>
+        {showSearch ? (
```
Comment (juancarlosfarah) [inline src/components/common/Header.js:46]: Maybe use: `showSearch &&`?

File: src/components/common/SearchSpaceBar.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+import { withRouter } from 'react-router';
+import PropTypes from 'prop-types';
+import InputBase from '@material-ui/core/InputBase';
+import { fade, withStyles } from '@material-ui/core/styles';
+import { withTranslation } from 'react-i18next';
+import SearchIcon from '@material-ui/icons/Search';
+import { setSearchQuery } from '../../actions';
+import { SPACE_SEARCH_INPUT_ID } from '../../config/selectors';
+
+const styles = theme => ({
```
Comment (juancarlosfarah) [inline src/components/common/SearchSpaceBar.js:12]: Does all this styling make it responsive to all cases?

**Comment by juancarlosfarah** [comment]
Can you please add a screenshot @pyphilia ?

File: src/reducers/SpaceReducer.js [hunk 5]
```diff
@@ -36,6 +38,10 @@ const INITIAL_STATE = Map({
     content: Set(),
     activity: false,
   }),
+  search: Map({
```
Comment (juancarlosfarah) [inline src/reducers/SpaceReducer.js:41]: I really like this! 👍

File: src/utils/search.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+// eslint-disable-next-line import/prefer-default-export
+export const searchSpacesByQuery = (spaces, query) => {
+  return spaces.filter(({ name }) =>
+    name.toLowerCase().includes(query.toLowerCase())
```
Comment (juancarlosfarah) [inline src/utils/search.js:4]: Maybe you want to also `trim` leading/trailing whitespace?

File: test/spaces/loadSpace.test.js [hunk 7]
```diff
@@ -92,8 +98,11 @@ describe('Load Space Scenarios', function() {
         await client.setValue(`#${LOAD_INPUT_ID}`, 'somefilepath');
         await client.pause(INPUT_TYPE_PAUSE);
 
-        const loadButtonHtml = await client.getHTML(`#${LOAD_LOAD_BUTTON_ID}`);
-        expect(loadButtonHtml).to.include('disabled');
+        const IsLoadButtonDisabled = await client.getAttribute(
```
Comment (juancarlosfarah) [inline test/spaces/loadSpace.test.js:101]: Not `isLoadButtonDisabled`?

File: test/spaces/loadSpace.test.js [hunk 8]
```diff
@@ -39,20 +39,26 @@ import { userSignIn } from '../userSignIn.test';
...
 
-  const value = await client.getValue(`#${LOAD_INPUT_ID}`);
+  const value = await client.getValue(loadInput);
   expect(value).to.equal(absolutePath);
 
   await client.click(`#${LOAD_LOAD_BUTTON_ID}`);
   await client.pause(LOAD_SPACE_PAUSE);
+};
+
+/* eslint-disable-next-line import/prefer-default-export */
```
Comment (juancarlosfarah) [inline test/spaces/loadSpace.test.js:57]: We don't need this line anymore, right?

**Comment by pyphilia** [comment]
Yes sorry, here the search bar in the home page
image

image

The same searchbar is in the spaces nearby
image

File: test/spaces/loadSpace.test.js (hunk 7 above)
Comment (pyphilia) [reply test/spaces/loadSpace.test.js:101]: I don't understand your comment?

**Pull request merged**